### PR TITLE
Fix bug in um.FieldsFileVariant output (no test yet).

### PR DESCRIPTION
**Pull request by pp-mo**
Fixes a bug in iris.experimental.um.FieldsFileVariant (investigating support issue).
The effect is that for files over a certain size, the lookup headers begin to overwrite the data payloads
It doesn't happen for smaller files, because the data-start is rounded up to the next block boundary.

To see the problem:
```
ff_file = um.FieldsFileVariant(test_filepath, mode=um.FieldsFileVariant.UPDATE_MODE)
ff_file.close()
```
This replaces the file with a re-written copy, which is corrupted if the file contains too many fields.

This PR is effectively just a heads-up and placeholder, because it should also contain a test, which I don't currently have time to do.

### CODE REVIEW

File: lib/iris/experimental/um.py [hunk 1]
```diff
@@ -590,12 +590,16 @@ def _update_fixed_length_header(self):
...
+            if len(set(header_lengths)) != 1:
+                msg = 'Inconsistent header lengths - {}'.format(header_lengths)
                 raise ValueError(msg)
-            header.lookup_shape = (lengths.pop(), len(self.fields))
+            header_length = header_lengths.pop()
+            n_fields = len(self.fields)
+            header.lookup_shape = (header_length, n_fields)
 
+            # make space for the lookup
+            word_number += header_length * n_fields
```
Comment (rhattersley) [inline lib/iris/experimental/um.py]: This line is the only real change, right?

File: lib/iris/experimental/um.py [hunk 2]
```diff
@@ -590,12 +590,16 @@ def _update_fixed_length_header(self):
         # Now deal with the LOOKUP and DATA components.
         if self.fields:
             header.lookup_start = word_number
-            lengths = set(field.num_values() for field in self.fields)
-            if len(lengths) != 1:
-                msg = 'Inconsistent header lengths - {}'.format(lengths)
+            header_lengths = [field.num_values() for field in self.fields]
+            if len(set(header_lengths)) != 1:
+                msg = 'Inconsistent header lengths - {}'.format(header_lengths)
```
Comment (rhattersley) [inline lib/iris/experimental/um.py]: - There's no need to switch `lengths` from being a `set` to being a `list`. (NB. A set supports `pop()`.)
- Given the ambiguity of the word "header" in the context of a FieldsFile variant, adding a `header_` prefix is *less* clear than just `lengths`. Perhaps `unique_num_values`? And then `num_values = unique_num_values.pop()`?

**Comment by rhattersley** [comment]
Thanks for spotting this @pp-mo - it must have been tricky to track down!

File: lib/iris/experimental/um.py (hunk 1 above)
Comment (pp-mo) [inline lib/iris/experimental/um.py]: Yup

File: lib/iris/experimental/um.py (hunk 2 above)
Comment (QuLogic) [inline lib/iris/experimental/um.py]: BTW, since iris requires Python 2.7, you could use a set comprehension here instead.

**Comment by pp-mo** [comment]
Thanks all.
Done now, with new test.
Please re-review @rhattersley

**Comment by rhattersley** [comment]
Looks good - thanks @pp-mo. I ran the test without your corresponding fix and it fell over nicely. :+1:

**Comment by rhattersley** [comment]
I'm happy to merge once there's a green tick.

**Pull request merged**